cql3/selection/scalar_function_selector.hh

```diff
@@ -13,13 +13,19 @@
 
 #include "abstract_function_selector.hh"
 #include "cql3/functions/scalar_function.hh"
+#include "cql3/functions/user_function.hh"
 #include "cql_serialization_format.hh"
+#include "lang/wasmtime.hh"
 
 namespace cql3 {
 
 namespace selection {
 
 class scalar_function_selector : public abstract_function_selector_for<functions::scalar_function> {
+    // auxiliary variables for reusing instances of wasm UDFs
+    std::optional<wasmtime::Store> _store;
+    std::optional<wasmtime::Instance> _instance;
+    std::optional<wasmtime::Func> _func;
```

Comment:
I now think this is the wrong place. First, it entangles wasm into a more generic context. Second, it doesn't allow reuse in single-row queries.

We should have a cache of instances, with eviction based on idle time or memory use. Execution should remove an entry from the cache, execute the function, and insert it into the cache again.